### PR TITLE
Feature - grant resource access to a specific resource group

### LINKED ISSUE #40: Provide script to grant a specific role to a resource on a specific resource group
**Is your feature request related to a problem? Please describe.**
In some cases a resource needs to be granted access to all resources present within a specific resource group. To help in setting this up, a PS-script can be provided.

**Describe the solution you'd like**
This script should allow you to define:
- the name of the resource (and its RG) which should be granted access
- the name of the resource group to which access should be granted
- the name of the role to assign

for instance:
```
New-ResourceGroupRoleAssignment
 -TargetResourceGroupName
 -ResourceName
 -ResourceGroupName
 -RoleDefinitionName
```

**Additional context**
This script can be created based on the following:
https://codit.visualstudio.com/Invictus%20Templates/_git/invictus-templates?path=%2Fdeploy%2FInvictus-FlowHandlerAccessRights.ps1

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add a PS module function member to grant a resource access to a specific resource group by assigning a role.

Closes #40

### CODE REVIEW

File: docs/preview/features/powershell/azure-security.md [hunk 1]
```diff
@@ -81,3 +81,22 @@ PS> Write-Host "Current SubscriptionId:" $Global:subscriptionId
 PS> Write-Host "Current AccessToken:" $Global:accessToken
 # Current AccessToken: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
 ```
+
+## Granting a resource access to all resources within a specific resource group
+
+In some cases a resource needs to be granted access to all resources present within a specific resource group.
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-security.md:87]:
```suggestion
In some cases, a resource needs to be granted access to all resources present within a specific resource group.  
This function allows you to assign an Azure built-in role to a resource upon a resource group.
```

File: docs/preview/features/powershell/azure-security.md [hunk 2]
```diff
@@ -81,3 +81,22 @@ PS> Write-Host "Current SubscriptionId:" $Global:subscriptionId
...
+
+| Parameter                 | Mandatory  | Description                                                                                  |
+| ------------------------- | ---------- | -------------------------------------------------------------------------------------------- |
+| `TargetResourceGroupName` | yes        | The name of the resource group to which access should be granted.                            |
+| `ResourceGroupName`       | yes        | The name of the resource which should be granted access.                                     |
```
Comment (mbraekman) [inline docs/preview/features/powershell/azure-security.md:92]: It seems the description for these 2 parameters (ResourceGroupName and ResourceName) has been switched around.

File: src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1 [hunk 3]
```diff
@@ -34,4 +34,34 @@ function Get-AzCachedAccessToken {
...
+  Grant a resource access to all resources present within a specific resource group.
+
+ .Parameter TargetResourceGroupName
+  The name of the resource group to which access should be granted.
+
+ .Parameter ResourceName
+  The name of the resource which should be granted access.
+
+ .Parameter ResourceGroupName
+  The name of the resource group where the resource is located which should be granted access.
```
Comment (mbraekman) [inline src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1:53]: Can you switch position with these 2 parameters -> ResourceGroupName, ResourceGroup

**Review by mbraekman — COMMENTED** [review]
Only some small corrections on the level of documentation are required.
Otherwise, good to go.

File: docs/preview/features/powershell/azure-security.md (hunk 2 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/azure-security.md:93]: Ah, yes, sorry. Good catch!

File: src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1 (hunk 3 above)
Comment (stijnmoreels) [reply src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1:53]: Yes, of course, they seem to be switched in the process.

File: docs/preview/features/powershell/azure-security.md (hunk 2 above)
Comment (stijnmoreels) [reply docs/preview/features/powershell/azure-security.md]: Done that. Resolving.

File: src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1 (hunk 3 above)
Comment (stijnmoreels) [reply src/Arcus.Scripting.Security/Arcus.Scripting.Security.psm1]: Done that. Resolving.

**Review by mbraekman — APPROVED** [review]
Looks good!

**Pull request merged**